auctions/english-auction/src/abi.sw

```diff
@@ -0,0 +1,23 @@
+library abi;
+
+dep data_structures;
+
+use data_structures::{Asset, Auction};
+use std::{contract_id::ContractId, identity::Identity, option::Option};
+
+abi EnglishAuction {
+    // #[storage(read)]fn auction_info(auction_id: u64) -> Option<Auction>;
+    #[storage(read, write)]fn bid(auction_id: u64, asset: Asset);
+    #[storage(read, write)]fn cancel_auction(auction_id: u64);
+    #[storage(read, write)]fn constructor(seller: Identity, sell_asset: Asset, buy_asset: Asset, inital_price: u64, reserve_price: u64, time: u64) -> u64;
+    // #[storage(read)]fn deposits(identity: Identity, auction_id: u64) -> Option<Asset>;
+    #[storage(read, write)]fn withdraw(auction_id: u64);
+    #[storage(read)]fn total_auctions() -> u64;
+}
+
+abi NFT {
```

Comment:
I cannot handle native tokens and NFTs the same. For native tokens I can check the message amount, but for NFTs I have to interact with the NFT contract and transfer it to the auction contract. It cannot be included in the transaction. For that, I need to know the abi of  the NFT and the functions I would like to call. Is there a way for me to import without defining in the abi.sw and then using `dep abi` in my utils.sw file?

Reply:
There's talk of being able to call contracts without knowing much about them. I do not recall the specifics though. I think using the "call" opcode, similar to what @matt-user has recently done in the DAO might be useful. I have mentioned that his code should be added to the std so perhaps once that is done then you can use it?

I haven't looked into it so matt can comment more on this.

Reply:
I don't know if there is a way to import without defining the abi.sw and using dep abi.  If you want to call a function on a contract not known at compile time you can use the call opcode specified [here](https://github.com/FuelLabs/fuel-specs/blob/1be31f70c757d8390f74b9e1b3beb096620553eb/specs/vm/instruction_set.md#call-call-contract).  However, currently I do not know of a way to encode the function selector and function arguments with the sdk, which is necessary to give the contract what it needs to use the call opcode.  However, it seems like you know your abi at compile time so it might be easiest just to use the NFT abi.